engine/CMakeLists.txt

```diff
@@ -1339,31 +1339,11 @@ ELSEIF (CMAKE_SYSTEM_NAME STREQUAL Linux)
 
     SET(VEGA_STRIKE_PYTHON_VERSION_STR "py3")
 
-    # lsb_release -i --> but then it'll find `Ubuntu` instead of Debian so will have to map more
-    # we can probably more reliably detect via checking for different packager commands
-    # `lsb_release` doesn't always work correctly, so it's easier to read the underlying file
-    # if it exists. It doesn't always exist. Debian doesn't use it, but Ubuntu does.
-    SET(LSB_EXISTS FALSE)
-    SET(LSB_LINUX_DISTRIBUTION "Unknown")
-    SET(LSB_LINUX_DISTRIBUTION_CODENAME "Unknown")
-
-    # Lookup the LSB Data - this is known for Ubuntu, but also used by other distros too.
-    # and may help with distro-sub-selection
-    FIND_PROGRAM(LSB_RELEASE_EXEC lsb_release)
-    IF (LSB_RELEASE_EXEC)
-        EXECUTE_PROCESS(COMMAND ${LSB_RELEASE_EXEC} --codename --short
-            OUTPUT_VARIABLE LSB_LINUX_DISTRIBUTION_CODENAME
-            OUTPUT_STRIP_TRAILING_WHITESPACE)
-        EXECUTE_PROCESS(COMMAND ${LSB_RELEASE_EXEC} --id --short
-            OUTPUT_VARIABLE LSB_LINUX_DISTRIBUTION
-            OUTPUT_STRIP_TRAILING_WHITESPACE)
-        SET(LSB_EXISTS TRUE)
-        MESSAGE("Found Linux Distribution Release Name: ${LSB_LINUX_DISTRIBUTION_CODENAME}")
-    ENDIF (LSB_RELEASE_EXEC)
+	# Detect available Linux Distros that can be built for
+	FIND_PACKAGE(LinuxDistro REQUIRED)
```

Comment:
Hmm. Not sure if I like the `FIND_PACKAGE` semantics here. Almost makes it sound like the Linux Distro is just another dependency, like SDL or expat. It's not. It's the entire operating system.

Could we change the usage to something like `INCLUDE(LinuxDistroDetect)` ?

Reply:
I modeled it after the other Find* dependencies; so it's following the same pattern.

Also, per the documentation FindPackage has many more options and would even give us the ability to pass parameters (via COMPONENTS) if we needed to.
- https://cmake.org/cmake/help/latest/command/find_package.html
- https://cmake.org/cmake/help/latest/command/include.html 

So I get what you're saying, but I think FIND_PACKAGE is still a better tool for the long run. I only did a few things we currently do now; if we start adding more distros then we may want to start adding options to it.

Reply:
OK, fair enough.